lib/vector-core/src/schema/requirement.rs

```diff
@@ -14,7 +14,7 @@ use super::Definition;
 #[derive(Debug, Clone, PartialEq)]
 pub struct Requirement {
     /// Semantic meanings configured for this requirement.
-    meaning: BTreeMap<&'static str, SemanticMeaning>,
+    meaning: BTreeMap<String, SemanticMeaning>,
```

Comment:
Would it be useful to use a `Cow<'static, str>` to support both here? Is this a mapping that is manipulated frequently, or just created once and looked up frequently?

Reply:
Based on a quick audit, I don't see this being in any critical path. Specifically, it is used to instantiate `schema::Requirement` structs. Then, it is looked up only in one [place](https://github.com/vectordotdev/vector/blob/master/src/topology/schema.rs#L326). I _assume_ this won't have any significant performance impact. 

Regardless, happy to refactor this to `Cow`. Let me know what you think.

Reply:
No, if it's not critical path then do whatever is simplest (which is _not_ `Cow`).